### ISSUE TITLE
SHAP for Motion Prediction in Autonomous Driving

### ISSUE THREAD

**Issue by Cram3r95** [shap/shap#2522]
Hi folks!

I am exploring this world of XAI in order to adapt it to Motion Prediction for Autonomous Driving. My inputs are map features (such as lanes, nodes, etc.) and a sequence of past trajectories of the agents, and the output is the future trajectory of the agent (s) of interest. Since it is a Deep Learning model, I have seen that DeepExplainer could fit, but on the other hand, KernelSHAP is defined as model agnostic.

In summary, I want to evaluate the contribution of the corresponding inputs to the final prediction, given a DL model and the input features can slightly change. What do you recommend me?

An example of Motion Prediction model is LaneGCN: 

https://github.com/uber-research/LaneGCN

![imagen](https://user-images.githubusercontent.com/15740838/165845759-d9916257-b904-4b4f-a203-dcc3566e9cfa.png)

**Comment by GT-111** [comment]
Hi, Cram!

Have you come up with some ideas? I'm currently working in this field too. What I want to do is similar to you. My inputs are feature maps and embedded past trajectories.

**Comment by Cram3r95** [comment]
Hi @GT-111 I left this topic, but probably I will move again after my PhD. Studying XAI for the different modules is quite interesting.
